.github/workflows/unit-test-runs.yml

```diff
@@ -0,0 +1,64 @@
+name: unit-test-standalone
+on:
+  push:
+  pull_request:
+
+env:
+  BACKWARDS_COMPATIBILITY_ARRAYS: OFF
+  TILEDB_S3: OFF
+  TILEDB_AZURE: OFF
+  TILEDB_GCS: OFF
+  TILEDB_SERIALIZATION: OFF
+  TILEDB_STATIC: OFF
+  TILEDB_TOOLS: ON
+  CXX: g++
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, ubuntu-latest]
+    name: Build - ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: 'Print env'
+        run: |
+          echo "'uname -s' is:"
+          echo "uname: " $(uname)
+          echo "uname -m: " $(uname -m)
+          echo "uname -r:" $(uname -r)
+          echo "uname -s: " $(uname -s)
+          echo "uname -v: " $(uname -v)
+          printenv
+        shell: bash
+
+      - name: 'Install system headers (OSX 10.14 only)'
+        run: |
+          set -e pipefail
+          open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
+          sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -allowUntrusted -target /
+        shell: bash
+        if: ${{ runner.os == 'macOS' && env.imageName == 'macOS-10.14' }}
+
+      - name: 'Build and run interval unit test'
+        id: test
+        run: |
+
+          mkdir -p $GITHUB_WORKSPACE/build
+          pushd $GITHUB_WORKSPACE/build
+
+          cmake .. \
```

Comment:
I haven't tried it yet myself (expediency), but one of the ideas of unit-local CMakeLists.txt was that they could be configured locally, not only through the top level. We haven't factored out our project-wide compile options so that they could be included in a single line within the local CMakeLists.txt, so currently it wouldn't yield tests in an identical environment as the one the library is built in.

I'd recommend doing this, but I've got no particular opinion on whether that belongs in this PR or a future one. We should, however, establish a good pattern early.

This also relates to the point about extraneous environment variables above.

Reply:
I looked at making the the unit test buildable directly this morning but its going to take a bit of work. We have to refactor out all common build flags as mentioned but also the superbuild functionality since the test requires catch. For this first pass it'll be easier to just require it to be built after the superbuild and I'll create a story and have someone make the changes necessary to have common cmake files that everything can inherit to break it out.